core-bundle/src/Resources/contao/models/PageModel.php

```diff
@@ -299,6 +301,34 @@ class PageModel extends Model
 	 */
 	protected $blnDetailsLoaded = false;
 
+	public function __set($strKey, $varValue)
+	{
+		if (\in_array($strKey, array('pageTitle', 'description', 'robots'), true))
+		{
+			trigger_deprecation('contao/core-bundle', '4.12', sprintf('Overriding "%s" is deprecated and will not work in Contao 5.0 anymore. Use the ResponseContext instead.', $strKey));
```

Comment:
I am not sure about this deprecation. What's deprecated is overriding these values for the pre-existing _global_ page object. But something like

```php
$newPage = new PageModel();
$newPage->pageTitle = 'foo';
```

should not trigger a deprecation, should it?

Reply:
True, but there's no way for me to know what you're going to do :D

Reply:
May be add `$GLOBALS['objPage'] ?? null === $this` to the condition (not sure if that works, but you get the idea).

Reply:
This works indeed. Fixed in 83bbaccf31b47001288e80cfa1072f9e5cfe12d7. Will mark the discussion as resolved.